common/common/shifts.py

```diff
@@ -0,0 +1,83 @@
+
```

Comment:
Applies to this file as a whole: tabs not spaces please! For consistency with the rest of the project if nothing else.